### ISSUE TITLE
GDScript 2.0: `ResourceSaver.save()` with using combined `SaverFlags` shows an invalid warning `INT_AS_ENUM_WITHOUT_CAST`

### ISSUE THREAD

**Issue by MikeSchulze** [godotengine/godot#74593]
### Godot version

v4.0.stable.official [92bee43ad]

### System information

Windows 10, MacOS

### Issue description

I actually fixing all the warnings that popup in my plugin after remove the default exclude script checks for the addons folder.

Using `ResourceSaver.save` says:
`The flags bitmask can be specified to customize the save behavior using SaverFlags flags.`

But let the compiler do report incorrect warnings
using the flags as bitmask (`ResourceSaver.FLAG_BUNDLE_RESOURCES|ResourceSaver.FLAG_REPLACE_SUBRESOURCE_PATHS`) is popup warnings.

```
[Ignorieren]Zeile 10 (INT_AS_ENUM_WITHOUT_CAST):Integer used when an enum value is expected. If this is intended cast the integer to the enum type.
[Ignorieren]Zeile 10 (INT_AS_ENUM_WITHOUT_MATCH):Cannot pass 66 as Enum "ResourceSaver.SaverFlags": no enum member has matching value.
```

### Steps to reproduce

```gdscript
ResourceSaver.save(GDScript.new(), "res://test.gd", ResourceSaver.FLAG_BUNDLE_RESOURCES|ResourceSaver.FLAG_REPLACE_SUBRESOURCE_PATHS)
```

### Minimal reproduction project

N/A

**Comment by dalexeev** [comment]
I think the problem is due to the signature of the [ResourceSaver.save()](https://docs.godotengine.org/en/stable/classes/class_resourcesaver.html#class-resourcesaver-method-save) method. For example in [Object.connect()](https://docs.godotengine.org/en/stable/classes/class_object.html#class-object-method-connect), the `flags` parameter is `int`, not the enum.

**Comment by dalexeev** [comment]
Closed by #75691 since I copied this small fix in the PR. #74641 renamed.

**Comment by dalexeev** [comment]
Closed for both master and 4.0.
